src/main/java/com/hivemq/cli/mqtt/AbstractMqttClientExecutor.java

```diff
@@ -158,6 +149,8 @@ public void disconnect(final @NotNull Disconnect disconnect) {
                     mqtt3Disconnect((Mqtt3Client) client, disconnect);
                     break;
             }
+
+            clientKeyToClientData.remove(disconnect.getKey());
```

Comment:
what happens, when the mqtt Disconnects throw an exception? This would mean, that the client is still in the map.

Reply:
good catch - I somehow forgot to remove this line.
The logic to remove the clients from the map on disconnects moved into the `ContextClientDisconnectListener` class. The logic should be fine there.